include/glow/Graph/Node.h

```diff
@@ -257,6 +261,13 @@ class Node : public Named,
   /// \returns the n'th result of the node.
   const NodeValue getNthResult(unsigned idx) const;
 
+  /// \returns the function holding this node.
+  /// If that node does not belong to any function, this
+  /// is nullptr.
+  Function *getParent() const { return parent_; }
+  /// Set the link to the function that holds this node.
+  void setParent(Function *parent) { parent_ = parent; }
+
```

Comment:
I wonder if `setParent` should really be a public function. 

IMHO, it could be safer if it would be private and Graph is a friend of Node and can invoke setParent. If it stays public it is possible to set it without actually moving the node to the nodes list of the parent function.

Reply:
I've followed what Instruction does. We also have a public setParent in the class Instruction and it is not protected at all :).

Given @nadavrot comment on NodeHandle, I don't think we want the extract protection.
In particular, notice that I've updated the Node::verify and that will fail if we attach a node to a function that it didn't belong too.

Reply:
Ah, OK. Never mind then ;-)